src/encryption.rs

```diff
@@ -77,13 +77,17 @@ impl From<&EncryptionSettings> for matrix_sdk_crypto::olm::EncryptionSettings {
 /// An encryption algorithm to be used to encrypt messages sent to a
 /// room.
 #[wasm_bindgen]
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialEq, Eq)]
 pub enum EncryptionAlgorithm {
     /// Olm version 1 using Curve25519, AES-256, and SHA-256.
     OlmV1Curve25519AesSha2,
 
     /// Megolm version 1 using AES-256 and SHA-256.
     MegolmV1AesSha2,
+
+    /// Unknown algorithm (JavaScript should ignore this value if it is
+    /// received, and should never send this value to Rust)
```

Comment:
From the point of view of a reader of this documentation, the fact that the library happens to be implemented in Rust is an implementation detail.

Also: it's not that we don't know what algorithm the device uses; rather it's that we don't support it.

```suggestion
    /// Unsupported algorithm.
    ///
    /// Applications should ignore this value if it is received, and should never set it.
```